ibis/expr/datatypes/core.py

```diff
@@ -642,6 +646,34 @@ def from_unit(cls, unit, timezone=None, nullable=True):
             raise ValueError(f"Invalid unit {unit}")
         return cls(scale=scale, timezone=timezone, nullable=nullable)
 
+    @classmethod
+    def from_datetime(
+        cls, dt: pydatetime.datetime, timezone: str | None = None, nullable: bool = True
+    ):
+        """Infer from a python datetime.datetime object."""
+        if dt.microsecond:
+            scale = 6
+        else:
+            scale = 0
+        if timezone is None and dt.tzinfo is not None:
+            timezone = str(dt.tzinfo)
```

Comment:
What's the rationale for having a `timezone` argument *and* inferring it from the input?

Reply:
eh, it's not needed, I wanted the API to follow the norm of "infer from the passed in object,but you can override if you want". I removed the argument though to keep it simpler. The user now needs to set the timezone on the passed in datetime if they want that. If in the future, we see that in many callsites the caller is needing to do this `if dt.tzinfo is None`... boilderplate, then we shoudl consider pushing this logic back down here.